src/edge/simulator.ts

```diff
@@ -270,22 +376,31 @@ export class Simulator {
             Constants.inputNamePrompt, null, "input1,input2");
     }
 
+    private async validateSimulatorInstalled(outputChannel: vscode.OutputChannel = null): Promise<InstallReturn> {
+        const telemetryName = "simulatorInstalled";
+        if (await this.simulatorInstalled()) {
+            return InstallReturn.Success;
+        } else {
+            TelemetryClient.sendEvent(`${telemetryName}.install`);
+            const installRes = await this.autoInstallSimulator(outputChannel);
+            TelemetryClient.sendEvent(`${telemetryName}.install.${InstallReturn[installRes]}`);
+            return installRes;
+        }
+    }
+
     private async callWithInstallationCheck(outputChannel: vscode.OutputChannel, callback: () => Promise<any>): Promise<any> {
-        const installReturn = await Simulator.validateSimulatorInstalled(outputChannel);
+        const installReturn = await this.validateSimulatorInstalled(outputChannel);
 
         switch (installReturn) {
             case InstallReturn.Success:
                 return await callback();
-            case InstallReturn.NoPip:
-                outputChannel.appendLine(Constants.outputNoSimulatorMsg);
-                throw new LearnMoreError(Constants.pipNotFoundMsg, Simulator.learnMoreUrl);
             case InstallReturn.Failed:
-                outputChannel.appendLine(Constants.outputNoSimulatorMsg);
-                throw new LearnMoreError(Constants.installFailedMsg, Simulator.learnMoreUrl);
+                await vscode.window.showErrorMessage(Constants.installStandaloneSimulatorFailedMsg);
```

Comment:
why remove the output channel message?

Reply:
No learn more message?

Reply:
Yes, learn more page is to help user to install pip package, I think we do not need it here.

Reply:
We also do not use output chanel here, because in this part, we cannot get deatailed error message. We already use output channel when error occured in "autoInstallSimulator" function